cep-XXXX.md

```diff
@@ -0,0 +1,122 @@
+   well; either as part of the same package or one of its dependencies.
+
+2. If the DSO was in `CONDA_BUILD_SYSROOT` (this is unset in most scenarios), then   these are system libraries and are allowed.
+
+3. If the DSO was in `C:\Windows\System32`, then these are system libraries and are
+   allowed.
+
+4. If the DSO was in the `DEFAULT_WIN_WHITELIST` location defined by the build tool,
+   then these are system libraries and are allowed. `DEFAULT_WIN_WHITELIST` is a
+   hardcoded constant in `conda-build` and `rattler-build`.
+
+Note that for Linux and macOS, the user has a way of controlling which system
+DSOs are allowed. On Linux, a user can modify the `sysroot_linux-*` package
+and on macOS, the user can specify a custom macOS SDK using `CONDA_BUILD_SYSROOT`.
+However, the user has no way of controlling which system DSOs are allowed
+on Windows.
+
+### Need for customizability of allowed Windows system libraries
+
+A user of conda build tools might need to customize the set of system DLLs.
+An example from Linux is the [removal of `libnsl` from `glibc`](https://github.com/conda-forge/linux-sysroot-feedstock/pull/40).
+
+In this case, newer versions of `glibc` dropped `libnsl` and made it a separate package,
+therefore conda-forge dropped it from `sysroot_linux-*` packages to make sure
+when new conda packages link to the `libnsl` system library, the build tool will
+warn or error out.
+
+An example for Windows can be found in this issue:
+[Warn when linking to debug VC runtimes](https://github.com/conda/conda-build/issues/5732).
+
+In this case, the CI machines provide debug VC (Visual Studio) runtime DLLs and are
+considered system DLLs by conda build tools, but these debug DLLs
+are not always found in user systems. Also, linking to these debug DLLs
+are not desirable and therefore should not be shipped it the corresponding
+conda package (e.g. `vc14_runtime`). Therefore we need a way to tell the build tool
+which DLLs are allowed and which are not.
+
+## Specification
+
+This CEP proposes a mechanism to control the allowed system DLLs on Windows.
+When a file provided by a conda package in `host` or `build` exists in any of
+the following locations we use those list of files:
+
+- `<BUILD_PREFIX>/etc/conda-build/(allow|deny)list.d/<subdir>/<conda-package-name>.txt`
+
+- `<PREFIX>/etc/conda-build/(allow|deny)list.d/<subdir>/<conda-package-name>.txt`
+
+When no allow list is available, but only a deny list is available, a default `C:\Windows\System32\*.dll` glob is assumed.
+When no deny list is available and an allow list is available,
+an empty deny list is assumed. When neither are available,
+conda build tools should fall back to the current method (as of August 2025).
+
+Each list supports standard POSIX globbing syntax; i.e., the following is allowed:
+
+- `C:\Windows\System32\*.dll`
+- `C:\foo.dll`
+- `**\R.dll`
+
+Note that backslashes are allowed and only absolute paths are allowed.
```

Comment:
Do we expect any files in the $PREFIX to be whitelisted? One could allow relative paths, or $PREFIX substitution.

Reply:
Backslashes + globs might be a dangerous combination because we cannot escape characters then I suppose?